src/Workspaces/Core/Portable/Classification/SyntaxClassification/EmbeddedLanguagesClassifier.cs

```diff
@@ -22,7 +22,10 @@ public EmbeddedLanguagesClassifier(IEmbeddedLanguagesProvider languagesProvider)
         {
             _languagesProvider = languagesProvider;
             SyntaxTokenKinds =
-                languagesProvider.Languages.SelectMany(p => p.Classifier.SyntaxTokenKinds).Distinct().ToImmutableArray();
+                languagesProvider.Languages.Where(p => p.Classifier != null)
```

Comment:
Is it worth introducing some kind of ordering or priority concept here? Correct me if I'm wrong but this works now because DateTimeEmbeddedLanguage doesn't have a classifier, but if it did, or a new embedded language is added, this problem could resurface, right?

We don't need it till we need it, but might be worth a comment to warn future developers.

Reply:
So we do have an ordering, and I'll link to it later. The problem here was that the ordering was violated by having the datetime language jump right to the fallback case. Without it doing that, the actual ordering kicks in and we get the right behavior.

Reply:
Coming back to this.  Here's our ordering:

```
        protected AbstractEmbeddedLanguageEditorFeaturesProvider(EmbeddedLanguageInfo info) : base(info)
        {
            Languages = ImmutableArray.Create<IEmbeddedLanguage>(
                new DateAndTimeEmbeddedLanguageEditorFeatures(info),
                new RegexEmbeddedLanguageEditorFeatures(this, info),
                new FallbackEmbeddedLanguage(info));
        }
```